src/Helper/MyParcelCollection.php

```diff
@@ -467,7 +468,20 @@ public function setPdfOfLabels($positions = self::DEFAULT_A4_POSITION): self
                 )
                 ->sendRequest('GET', MyParcelRequest::REQUEST_TYPE_RETRIEVE_LABEL);
 
-            $this->label_pdf = $request->getResult();
+            /**
+             * When account needs to pay upfront, an array is returned with payment information,
+             * instead of the actual pdf’s. It will throw an unintelligible error when not handled here.
+             */
+            $result = $request->getResult();
+
+            if (!is_string($result) || !preg_match('/^%PDF-1./', $result)) {
+                if (is_array($result) && isset($result['data']['payment_instructions'])) {
+                    throw new ApiException('Received payment link instead of pdf. Check your MyParcel account status.');
```

Comment:
Shouldn't these messages be localized?

Reply:
Made a guild topic from this as discussed. For now it can be closed imo.